src/components/forms/ProxyDeviceForm.tsx

```diff
@@ -258,30 +277,58 @@ const ProxyDeviceForm: FC<Props> = ({ formik, project }) => {
 
     customRows.push(
       getConfigurationRowBase({
-        className: "no-border-top inherited-with-form",
+        className: "no-border-top inherited-with-form p-heading--6",
         configuration: <Label>Listen</Label>,
+      }),
+    );
+
+    customRows.push(
+      getConfigurationRowBase({
+        className: "no-border-top inherited-with-form",
+        configuration: <Label>Type</Label>,
+        inherited: (
+          <Select
+            onChange={(e) => {
+              ensureEditMode(formik);
+              const newType = e.target.value;
+              void formik.setFieldValue(
+                `devices.${index}.listen`,
+                `${newType}:${listenAddress}:${listenPort}`,
+              );
+
+              if (device.nat === "true") {
+                void formik.setFieldValue(
+                  `devices.${index}.connect`,
+                  `${newType}:${connectAddress}:${connectPort}`,
+                );
+              }
```

Comment:
We also need to set the connect type to one matching in listen. Otherwise, we have this problem:

1. user selects connect type tcp
2. user select listen type udp
3. user enables nat mode

Now the list type is wrong, but the user can't change it to match.

Reply:
Yep, that's what I'm working on now. :)

Reply:
Alrighty, I have addressed (and pushed) this now. Moving onto the other issues.